pkg/cmd/azurekeyvaulttemplatefuncs.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
+)
+
+const azureVaultDomain = "vault.azure.net"
+
+type azureKeyVault struct {
+	Name   string `json:"name"  mapstructure:"name"  yaml:"name"`
+	Alias  string `json:"alias" mapstructure:"alias" yaml:"alias"`
+	client *azsecrets.Client
+	cache  map[string]string
+}
+
+func (v *azureKeyVault) URL() string {
+	return "https://" + v.Name + "." + azureVaultDomain + "/"
+}
+
+type azureKeyVaultConfig struct {
+	Vaults []*azureKeyVault `json:"vaults" mapstructure:"vaults" yaml:"vaults"`
+	cred   *azidentity.DefaultAzureCredential
```

Comment:
I’ll go a step further than @twpayne about the removal of `alias:`, I would change this to:

```go
type azureKeyVaultConfig struct {
  DefaultVault String `json:"defaultVault" toml:"defaultVault" yaml:"defaultVault"`
  vaults map[string]*azureKeyVault
  cred *azidentity.DefaultAzureCredential
```

`azureKeyVault` can then be simplified to `name`, `client`, and `cache` with no serialization instructions. The `vaults` map is initialized with an entry for `DefaultVault`.

Reply:
Implemented.